Source/ARTRealtime.m

```diff
@@ -810,12 +811,28 @@ - (BOOL)isActive {
 - (void)sendImpl:(ARTProtocolMessage *)msg callback:(void (^)(ARTStatus *))cb {
     if (msg.ackRequired) {
         msg.msgSerial = [NSNumber numberWithLongLong:self.msgSerial++];
```

Comment:
This should be moved below the error handling, so that errors don't increase msgSerial.